cmdstanpy/stanfit.py

```diff
@@ -547,21 +560,93 @@ def get_drawset(self, params: List[str] = None) -> pd.DataFrame:
+        if self._drawset is None:
+            # pylint: disable=redundant-keyword-arg
+            data = self.sample.reshape(
+                (self.num_draws * self.runset.chains),
+                len(self.column_names),
+                order='A',
+            )
+            self._drawset = pd.DataFrame(data=data, columns=self.column_names)
         if params is None:
-            return drawset
+            return self._drawset
         mask = []
         for param in params:
             for name in self.column_names:
                 if param == name or param == name.split('.')[0]:
                     mask.append(name)
-        return drawset[mask]
+        return self._drawset[mask]
+
+    def stan_variable(self, name: str) -> np.ndarray:
+        """
+        Return a new ndarray which contains the set of draws
+        for the named Stan program variable.
+
+        * If the variable is a scalar variable, this returns a 1-d array,
+          length(draws X chains).
+        * If the variable is a vector, this is a 2-d array,
+          shape ( draws X chains, len(vector))
+        * If the variable is a matrix, this is a 3-d array,
+          shape ( draws X chains, matrix nrows, matrix ncols ).
+        * If the variable is an array with N dimensions, this is an N+1-d array,
+          shape ( draws X chains, size(dim 1), ... size(dim N)).
+
+        :param name: variable name
+        """
+        if name not in self._stan_var_dims:
+            raise ValueError('unknown name: {}'.format(name))
+        if self._drawset is None:
+            self.get_drawset()
+        dims = self._stan_var_dims[name]
+        if dims == 1:
+            return self._drawset[name]
+        else:
+            var_dims = [self._drawset.shape[0]]
+            var_dims.extend(dims)
+            var_cols = [
+                col for col in self._drawset if col.startswith(name + '.')
+            ]
+            return (
+                self._drawset[var_cols]
+                .to_numpy()
+                .reshape(tuple(var_dims), order='A')
+            )
+
+    def stan_variables(self) -> Dict:
+        """
+        Return a dictionary of all Stan program variables.
+        Creates copies of the data in the draws matrix.
+        """
+        result = {}
+        if self._drawset is None:
+            self.get_drawset()
+        for key in self.stan_var_dims:
+            dims = self._stan_var_dims[key]
+            if dims == 1:
+                result[key] = self._drawset[key]
+            else:
+                var_dims = [self._drawset.shape[0]]
+                var_dims.extend(dims)
+                var_cols = [
+                    col for col in self._drawset if col.startswith(key + '.')
+                ]
+                result[key] = (
+                    self._drawset[var_cols]
+                    .to_numpy()
+                    .reshape(tuple(var_dims), order='A')
+                )
+        return result
```

Comment:
This can call the other function. 

Not sure if the `self.get_drawset()`is needed or can it also be in the `stan_variable` or is `self._stan_var_dims` filled in `self.get_drawset()`

```suggestion
        result = {}
        if self._drawset is None:
            self.get_drawset()
        for name in self.stan_var_dims:
            result[name] = self.stan_variable(name)
        return result
```